From 74f67d6898d2862a0413160ac9095c6d34ebbbc1 Mon Sep 17 00:00:00 2001
From: Sheng Nan <b38800@freescale.com>
Date: Mon, 22 Oct 2012 18:39:39 +0800
Subject: [PATCH 358/505] ENGR00230653 Camera: modify doc for MIPI CSI capture
 and overlay test

- add camera input device choice "-d /dev/video1" for mipi csi
capture on smart device board.
- mipi csi support 2592*1944 @ 15fps only, add "-fr 15" for mxc_overlay_test

Signed-off-by: Sheng Nan <b38800@freescale.com>
---
 test/mxc_v4l2_test/doc/FSL-UT-V4L2-008.txt |  4 ++--
 test/mxc_v4l2_test/doc/FSL-UT-V4L2-009.txt | 38 ++++++++++++++++--------------
 2 files changed, 22 insertions(+), 20 deletions(-)

diff --git a/test/mxc_v4l2_test/doc/FSL-UT-V4L2-008.txt b/test/mxc_v4l2_test/doc/FSL-UT-V4L2-008.txt
index e7ea5f5..db8fcac 100644
--- a/test/mxc_v4l2_test/doc/FSL-UT-V4L2-008.txt
+++ b/test/mxc_v4l2_test/doc/FSL-UT-V4L2-008.txt
@@ -45,7 +45,7 @@ Test 5. mxc_v4l2_overlay.out -iw 1280 -ih 720 -ow 640 -oh 480 -m 4 -di /dev/vide
 
 Test 6. mxc_v4l2_overlay.out -iw 1920 -ih 1080 -ow 640 -oh 480 -m 5 -di /dev/video1 -fg
 
-Test 7. mxc_v4l2_overlay.out -iw 2592 -ih 1944 -ow 640 -oh 480 -m 6 -di /dev/video1 -fg
+Test 7. mxc_v4l2_overlay.out -iw 2592 -ih 1944 -ow 640 -oh 480 -m 6 -fr 15 -di /dev/video1 -fg
 
 Test background display:
 
@@ -61,7 +61,7 @@ Test 12. mxc_v4l2_overlay.out -iw 1280 -ih 720 -ow 640 -oh 480 -m 4 -di /dev/vid
 
 Test 13. mxc_v4l2_overlay.out -iw 1920 -ih 1080 -ow 640 -oh 480 -m 5 -di /dev/video1 -bg
 
-Test 14. mxc_v4l2_overlay.out -iw 2592 -ih 1944 -ow 640 -oh 480 -m 6 -di /dev/video1 -bg
+Test 14. mxc_v4l2_overlay.out -iw 2592 -ih 1944 -ow 640 -oh 480 -m 6 -fr 15 -di /dev/video1 -bg
 
 | Expected Result |
 The image from the camera is displayed on the screen. Colors and image should be correct.
diff --git a/test/mxc_v4l2_test/doc/FSL-UT-V4L2-009.txt b/test/mxc_v4l2_test/doc/FSL-UT-V4L2-009.txt
index 833cee6..5214b30 100644
--- a/test/mxc_v4l2_test/doc/FSL-UT-V4L2-009.txt
+++ b/test/mxc_v4l2_test/doc/FSL-UT-V4L2-009.txt
@@ -21,45 +21,45 @@ N/A
 ov5640 sensor and MIPI extention board are requried
 
 | Test Procedure |
-Check that /dev/video0 exists. If it does not exist, load the camera driver with the command: modprobe mxc_v4l2_capture
+Check that /dev/video1 exists. If it does not exist, load the camera driver with the command: modprobe mxc_v4l2_capture
 
-modprobe CAMERA_MODULE (CAMERA_MODULE should be replaced by the camera module on the related platform, e.g. ov5640_camera_mipi should be used here on MX6Q?).
+modprobe CAMERA_MODULE (CAMERA_MODULE should be replaced by the camera module on the related platform, e.g. ov5640_camera_mipi should be used here on MX6Q).
 
-Check again for /dev/video0. If it is not there, the camera driver has not loaded successfully and the test has failed.
+Check again for /dev/video1. If it is not there, the camera driver has not loaded successfully and the test has failed.
 
 Type the following command lines in a writable directory. The 50 frames of video from the camera will be captured in the file "test.yuv" for each resolution. The format will be YUV420.
 
 Test CSI->SMFC->MEM:
 
-Test 1. mxc_v4l2_capture.out -iw 640 -ih 480 -ow 640 -oh 480 -m 0 -i 1 -r 0 -c 50 -fr 30 test.yuv
+Test 1. mxc_v4l2_capture.out -iw 640 -ih 480 -ow 640 -oh 480 -m 0 -i 1 -r 0 -c 50 -fr 30 -d /dev/video1 test.yuv
 
-Test 2. mxc_v4l2_capture.out -iw 320 -ih 240 -ow 320 -oh 240 -m 1 -i 1 -r 0 -c 50 -fr 30 test.yuv
+Test 2. mxc_v4l2_capture.out -iw 320 -ih 240 -ow 320 -oh 240 -m 1 -i 1 -r 0 -c 50 -fr 30 -d /dev/video1 test.yuv
 
-Test 3. mxc_v4l2_capture.out -iw 720 -ih 480 -ow 720 -oh 480 -m 2 -i 1 -r 0 -c 50 -fr 30 test.yuv
+Test 3. mxc_v4l2_capture.out -iw 720 -ih 480 -ow 720 -oh 480 -m 2 -i 1 -r 0 -c 50 -fr 30 -d /dev/video1 test.yuv
 
-Test 4. mxc_v4l2_capture.out -iw 720 -ih 576 -ow 720 -oh 576 -m 3 -i 1 -r 0 -c 50 -fr 30 test.yuv
+Test 4. mxc_v4l2_capture.out -iw 720 -ih 576 -ow 720 -oh 576 -m 3 -i 1 -r 0 -c 50 -fr 30 -d /dev/video1 test.yuv
 
-Test 5. mxc_v4l2_capture.out -iw 1280 -ih 720 -ow 1280 -oh 720 -m 4 -i 1 -r 0 -c 50 -fr 30 test.yuv
+Test 5. mxc_v4l2_capture.out -iw 1280 -ih 720 -ow 1280 -oh 720 -m 4 -i 1 -r 0 -c 50 -fr 30 -d /dev/video1 test.yuv
 
-Test 6. mxc_v4l2_capture.out -iw 1920 -ih 1080 -ow 1920 -oh 1080 -m 5 -i 1 -r 0 -c 50 -fr 30 test.yuv
+Test 6. mxc_v4l2_capture.out -iw 1920 -ih 1080 -ow 1920 -oh 1080 -m 5 -i 1 -r 0 -c 50 -fr 30 -d /dev/video1 test.yuv
 
-Test 7. mxc_v4l2_capture.out -iw 2592 -ih 1944 -ow 2592 -oh 1944 -m 6 -i 1 -r 0 -c 50 -fr 15 test.yuv
+Test 7. mxc_v4l2_capture.out -iw 2592 -ih 1944 -ow 2592 -oh 1944 -m 6 -i 1 -r 0 -c 50 -fr 15 -d /dev/video1 test.yuv
 
 Test CSI->PRP->MEM:
 
-Test 8. mxc_v4l2_capture.out -iw 640 -ih 480 -ow 640 -oh 480 -m 0 -r 0 -c 50 -fr 30 test.yuv
+Test 8. mxc_v4l2_capture.out -iw 640 -ih 480 -ow 640 -oh 480 -m 0 -r 0 -c 50 -fr 30 -d /dev/video1 test.yuv
 
-Test 9. mxc_v4l2_capture.out -iw 320 -ih 240 -ow 640 -oh 480 -m 1 -r 0 -c 50 -fr 30 test.yuv
+Test 9. mxc_v4l2_capture.out -iw 320 -ih 240 -ow 640 -oh 480 -m 1 -r 0 -c 50 -fr 30 -d /dev/video1 test.yuv
 
-Test 10. mxc_v4l2_capture.out -iw 720 -ih 480 -ow 640 -oh 480 -m 2 -r 0 -c 50 -fr 30 test.yuv
+Test 10. mxc_v4l2_capture.out -iw 720 -ih 480 -ow 640 -oh 480 -m 2 -r 0 -c 50 -fr 30 -d /dev/video1 test.yuv
 
-Test 11. mxc_v4l2_capture.out -iw 720 -ih 576 -ow 640 -oh 480 -m 3 -r 0 -c 50 -fr 30 test.yuv
+Test 11. mxc_v4l2_capture.out -iw 720 -ih 576 -ow 640 -oh 480 -m 3 -r 0 -c 50 -fr 30 -d /dev/video1 test.yuv
 
-Test 12. mxc_v4l2_capture.out -iw 1280 -ih 720 -ow 640 -oh 480 -m 4 -r 0 -c 50 -fr 30 test.yuv
+Test 12. mxc_v4l2_capture.out -iw 1280 -ih 720 -ow 640 -oh 480 -m 4 -r 0 -c 50 -fr 30 -d /dev/video1 test.yuv
 
-Test 13. mxc_v4l2_capture.out -iw 1920 -ih 1080 -ow 640 -oh 480 -m 5 -r 0 -c 50 -fr 30 test.yuv
+Test 13. mxc_v4l2_capture.out -iw 1920 -ih 1080 -ow 640 -oh 480 -m 5 -r 0 -c 50 -fr 30 -d /dev/video1 test.yuv
 
-Test 14. mxc_v4l2_capture.out -iw 2592 -ih 1944 -ow 640 -oh 480 -m 6 -r 0 -c 50 -fr 15 test.yuv
+Test 14. mxc_v4l2_capture.out -iw 2592 -ih 1944 -ow 640 -oh 480 -m 6 -r 0 -c 50 -fr 15 -d /dev/video1 test.yuv
 
 | Expected Result |
 There should be no errors or warnings. The file "test.yuv" will be created.
@@ -80,7 +80,9 @@ For Test 6. mxc_v4l2_output.out -iw 1920 -ih 1080 -ow 640 -oh 480 -r 0 -fr 30 te
 
 For Test 7. mxc_v4l2_output.out -iw 2592 -ih 1944 -ow 640 -oh 480 -r 0 -fr 15 test.yuv
 
-For Test 8-14. mxc_v4l2_output.out -iw 640 -ih 480 -ow 640 -oh 480 -r 0 -fr 30 test.yuv
+For Test 8-13. mxc_v4l2_output.out -iw 640 -ih 480 -ow 640 -oh 480 -r 0 -fr 30 test.yuv
+
+For Test 14. mxc_v4l2_output.out -iw 640 -ih 480 -ow 640 -oh 480 -r 0 -fr 15 test.yuv
 
 |====================================================================
 
-- 
1.8.0

